client/src/app/users/user-list.component.spec.ts

```diff
@@ -39,11 +39,39 @@ const COMMON_IMPORTS: any[] = [
   RouterTestingModule,
 ];
 
-describe('UserListComponent', () => {
+// The `UserListComponent` being tested
+let userList: UserListComponent;
+
+// This constructs the `userList` (declared
+// above) that will be used throughout the tests.
+// This is called in a `beforeEach()` in each of the
+// `describe()` sections below.
+const setupUserList = () =>
+  // Compile all the components in the test bed
+  // so that everything's "ready to go".
+  TestBed.compileComponents().then(() => {
+    // Create a "fixture" of the UserListComponent. that
+    // allows us to get an instance of the component
+    // (userList, below) that we can "control" in
+    // the tests.
+    const fixture = TestBed.createComponent(UserListComponent);
+    userList = fixture.componentInstance;
+    // Tells Angular to "sync" the data bindings between
+    // the model and the DOM. This ensures, e.g., that the
+    // `userList` component actually requests the list
+    // of users from the `MockUserService` so that it's
+    // "up to date" before we start running tests on it.
+    // fixture.detectChanges();
+    fixture.detectChanges();
+  });
 
-  let userList: UserListComponent;
-  let fixture: ComponentFixture<UserListComponent>;
+describe('UserListComponent', () => {
 
+  // Set up the `TestBed` so that it "knows" that we're using
+  // a `MockUserService` in place of the "real" `UserService`
+  // for the purposes of the testing. We also have to include
+  // the relevant imports and declarations so that the tests
+  // can "find" all the necessary parts.
```

Comment:
Again, scare quotes are probably unnecessary 🙂 Don't worry, you're using all of these words in their normal senses!

```suggestion
  // Set up the `TestBed` so that it uses
  // a `MockUserService` in place of the real `UserService`
  // for the purposes of the testing. We also have to include
  // the relevant imports and declarations so that the tests
  // can find all the necessary parts.
```

Reply:
I agree 100%! Thanks for catching all that!